static/src/javascripts/projects/common/modules/commercial/build-page-targeting.ts

```diff
@@ -213,8 +214,73 @@ const getPageTargeting = (consentState: ConsentState): PageTargeting => {
 	return pageTargeting;
 };
 
+const consentlessTargetingKeys = [
+	'ab',
+	'at',
+	'bl',
+	'bp',
+	'br',
+	'category',
+	'cc',
+	'ct',
+	'dcre',
+	'edition',
+	'isbn',
+	'k',
+	'refreshed',
+	'rp',
+	's',
+	'se',
+	'sens',
+	'sh',
+	'si',
+	'skinsize',
+	'slot',
+	'slot-fabric',
+	'su',
+	'tn',
+	'url',
+	'urlkw',
+] as const;
+
+type ConsentlessTargetingKeys = typeof consentlessTargetingKeys[number];
+
+type ConsentlessPageTargeting = Partial<
+	Pick<PageTargeting, ConsentlessTargetingKeys>
+>;
+
+const isConsentlessKey = (key: unknown): key is ConsentlessTargetingKeys => {
+	return consentlessTargetingKeys.includes(key as ConsentlessTargetingKeys);
+};
+
+const getConsentlessPageTargeting = (
+	consentState: ConsentState,
+): ConsentlessPageTargeting => {
+	const consentedPageTargeting: PageTargeting =
+		getPageTargeting(consentState);
+
+	// filter consentedPageTargeting to only include consentless keys.
+	// A more straightforward way to do this would be to loop through
+	// consentedPageTargeting and only include keys that are in
+	// consentlessTargetingKeys. This causes type errors since TypeScript can't
+	// verify that the values are of the correct type.
+	// We use reduce to get around this.
+	const consentlessPageTargeting: ConsentlessPageTargeting = Object.entries(
+		consentedPageTargeting,
+	).reduce(
+		(consentlessPageTargeting, [key, value]) =>
+			isConsentlessKey(key)
+				? { ...consentlessPageTargeting, [key]: value }
+				: consentlessPageTargeting,
```

Comment:
Minor feel free to ignore:

- do we need a unit test or is simple enough here to not need a unit test?
- the reduce spreading and creating a new object each iteration is _probably_ pretty slow and a bit hard to read. What do you think of something like:

```
Object.fromEntries(Object.entries(consentedPageTargeting).filter(({k ,v}) => isConsentlessKey(key)))
```

Reply:
Thanks, this is much easier to read!